#!/usr/bin/env python
PACKAGE = "ball_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_hue", int_t, 0, "Minimum hue",   0, 0, 180)
gen.add("max_hue", int_t, 0, "Maximum hue", 180, 0, 180)

gen.add("min_sat", int_t, 0, "Minimum saturation",   0, 0, 255)
gen.add("max_sat", int_t, 0, "Maximum saturation", 255, 0, 255)

gen.add("min_val", int_t, 0, "Minimum brightness",   0, 0, 255)
gen.add("max_val", int_t, 0, "Maximum brightness", 255, 0, 255)

gen.add("gauss_size", int_t, 0, "Gauss filter size", 9, 0, 20)
gen.add("gauss_sigma", double_t, 0, "Gauss filter sigma", 2, 0.1, 20)

exit(gen.generate(PACKAGE, "ball_detection_node", "BallDetection"))

